driver/src/main/java/org/neo4j/driver/internal/connector/socket/TrustOnFirstUseTrustManager.java

```diff
@@ -156,11 +158,29 @@ public void checkServerTrusted( X509Certificate[] chain, String authType )
                         "in the file `%s`.\n" +
                         "The old certificate saved in file is:\n%sThe New certificate received is:\n%s",
                         serverId, serverId, knownCerts.getAbsolutePath(),
-                        X509CertToString( this.cert ), X509CertToString( cert ) ) );
+                        X509CertToString( this.fingerprint ), X509CertToString( cert ) ) );
             }
         }
     }
 
+    /**
+     * Calculate the certificate fingerprint - simply the SHA-1 hash of the DER-encoded certificate.
+     */
+    public static String fingerprint( X509Certificate cert ) throws CertificateException
+    {
+        try
+        {
+            MessageDigest md = MessageDigest.getInstance( "SHA-1" );
```

Comment:
How about let's move to SHA-2 directly as SHA-1 could already be reverted nowadays?